components/gemini/browser/gemini_json_parser.cc

```diff
@@ -0,0 +1,176 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "brave/components/gemini/browser/gemini_json_parser.h"
+
+#include "base/json/json_reader.h"
+
+bool GeminiJSONParser::GetTokensFromJSON(
+    const std::string& json, std::string *access_token,
+    std::string* refresh_token) {
+  if (!access_token || !refresh_token) {
+    return false;
+  }
+
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          json, base::JSONParserOptions::JSON_PARSE_RFC);
+  base::Optional<base::Value>& records_v = value_with_error.value;
```

Comment:
you are missing a header for base/values.h

Reply:
fixed